CMSgov/managed-care-review
# MR-2233: State can add certifying actuary on rate details page

## Summary
[MR-2233](https://qmacbis.atlassian.net/browse/MR-2233)

- Add certifying actuary contact form for each rate certification section on the Rate Details page.
- Add `ActuaryContactFields` component and refactor `Contacts.tsx` to use his component.
- Update `rateInfos` domain model to include `actuaryContacts` and `actuaryCommunicationPreference` fields.
- Update tests for domain model changes.
- Update `hasValidRates` to check for valid actuary contacts data.
- Update `updateToVersion3()` to convert the second actuary contact and beyond to `addtlActuaryContacts`.

#### Related issues

#### Screenshots

#### Test cases covered

Unit tests
- `RateDetails.test.tsx`
   - `handles multiple rates`: Update tests with assertions for certifying actuary contact fields

- `healthPlanFormData.ts`:
   - `'hasValidRates evaluates as expected'`: Update tests to cover missing actuary contact fields

- `toLatestProtoVersion.test.ts`
   - `'should convert actuary contacts after the first to addtlActuaryContacts and communication preference as expected'`
   - `'should return empty array for addtlActuaryContacts when only a single actuary contact exists'`

- Cypress Tests
   - `contacts.spec.ts`: Update to fill out rate details before filling out contacts.
   - `fillOutNewRateCertification` and `fillOutAmendmentToPriorRateCertification`: Code to fill out certifying actuary contact fields



## QA guidance

**Because we are running `toLatestVersion()` based on proto version. These tests can only be done on older submission data with a previous proto version. I have created two submission on VAL (FL) that can be used to test these two scenarios in preparation for PO/Design review.**
[Single actuary contact](https://mc-review-val.onemac.cms.gov/submissions/16deca1b-0c97-4d39-9bd6-d766a1177169)
[Multiple actuary contacts](https://mc-review-val.onemac.cms.gov/submissions/ed68a272-47d5-4b28-bfa7-b76130e69b93)

- Old actuary data can be displayed correctly with feature flag on.
1. <strike>Turn `multi-rate-submissions` flag **off** and fill out a contract and rates submission with more then 1 actuary contact.</strike>
2. <strike>Submit, then unlock.</strike>
2. Turn feature flag on
3. Navigate to Rate details page and verify certifying actuary is correct for the first ( and only rate).
4. Click continue and verify additional actuary information is correct.
  
- Submissions with old actuary data can still be resubmitted when feature flag is off.
1. Turn `multi-rate-submissions` flag **off** and fill out a contract and rates submission with more then 1 actuary contact.
2. Submit, then unlock.
3. Edit actuary contacts on Contacts page and continue to submit.
